networkx/algorithms/similarity.py

```diff
@@ -1550,6 +1575,10 @@ def panther_similarity(
     """
     import numpy as np
 
+    G = G.subgraph(
+        [node for node in G.nodes if node not in list(nx.isolates(G))]
+    ).copy()
```

Comment:
Is this supposed to silently ignore the isloates? Currently a KeyError pops up if there is a isolate and if there is only one node a `ZeroDivisionError: float division by zero` pops up.

Code examples:
``` python
>>> G = nx.Graph()
>>> G.add_node(1)
>>> nx.panther_similarity(G, 1)
...
...
ZeroDivisionError: float division by zero

>>> G.add_edge(1, 2)
>>> G.add_node(3)
>>> nx.panther_similarity(G, 3)
....
....
KeyError: 3
```

Reply:
Thanks for the review! I have raised `NetworkXUnfeasible` error if the `source` node is an isolated node and it will also take care of the null graph case. And otherwise it will silently ignore the isolated nodes.

Reply:
Another way could be to only allow connected graphs(with more than 1 node) to be passed as `G`. Please let me know your thoughts on this!